src/pairs.c

```diff
@@ -163,154 +164,45 @@ bigistore_pairs_insert(BigIStorePairs *pairs, int32 key, int64 val)
 
 /*
  * return the number of digits
+ *
+ * See: https://math.stackexchange.com/questions/231742
  */
 static inline int
 digits32(int32 num)
 {
-    int sign = 0;
+    int digits = 1;
+
+    if (num == 0)
+        return 1;
     if (num < 0)
     {
         if (num == INT_MIN)
             // special case for -2^31 because 2^31 can't fit in a two's complement 32-bit integer
-            return 12;
-        sign = 1;
-        num  = -num;
+            return 11;
+        digits++;
+        num = -num;
     }
 
-    if (num < 1e5)
-    {
-        if (num < 1e3)
-        {
-            if (num < 10)
-                return 1 + sign;
-            else if (num < 1e2)
-                return 2 + sign;
-            else
-                return 3 + sign;
-        }
-        else
-        {
-            if (num < 1e4)
-                return 4 + sign;
-            else
-                return 5 + sign;
-        }
-    }
-    else
-    {
-        if (num < 1e7)
-        {
-            if (num < 1e6)
-                return 6 + sign;
-            else
-                return 7 + sign;
-        }
-        else
-        {
-            if (num < 1e8)
-                return 8 + sign;
-            else if (num < 1e9)
-                return 9 + sign;
-            else
-                return 10 + sign;
-        }
-    }
+    digits += floor(log10(num));
+    return digits;
```

Comment:
uh we should do some serious benchmarking on this
there is a reason why we went for this complex approach

Reply:
You're right. Here is a quick benchmark of two versions:
https://gist.github.com/zilder/fc2035aa8c8c78a32c71554917707911

```
./bench
checking digits_old... 12535 ms
checking digits_new... 41895 ms
```
So yes, it doesn't work as fast as old one. I'll revert these changes.